File: Stack/adjacent.py
# Remove Adjacent Duplicates from String
def adjacent(str):
    stack =[]

    for char in str:
        stackLength = len(stack)-1
        if stackLength >=0 and char == stack[stackLength]:
            stack.pop()
        else:
            stack.append(char)    
    return "".join(stack)




print(adjacent("abbaca"))

File: Stack/adjacent.js
// # Remove Adjacent Duplicates from String
function adjacent(str) {
    stack = []

    for (char of str) {
        stackLength = stack.length - 1;
        if (stackLength >= 0 && char == stack[stackLength]) {
            stack.pop()
        }
        else {
            stack.push(char)

        }
    }
    return stack.join("")
}


console.log(adjacent("abbaca"))